### PR TITLE
CLN: Add dataclass DerivedNamedStratigraphy

### PR DESCRIPTION
**Pull request by janbjorge**
Adressess: https://github.com/equinor/fmu-dataio/issues/457

### CODE REVIEW

File: src/fmu/dataio/_objectdata_provider.py [hunk 1]
```diff
@@ -155,25 +203,23 @@ class _ObjectDataProvider:
...
-    fmt: str = field(default="", init=False)
-    extension: str = field(default="", init=False)
-    layout: str = field(default="", init=False)
-    bbox: dict = field(default_factory=dict, init=False)
-    specs: dict = field(default_factory=dict, init=False)
-    time0: str = field(default="", init=False)
-    time1: str = field(default="", init=False)
-
-    def __post_init__(self) -> None:
-        logger.info("Ran __post_init__")
```
Comment (janbjorge) [inline src/fmu/dataio/_objectdata_provider.py:176]: Seems unnecessary to me.

File: src/fmu/dataio/_objectdata_provider.py [hunk 2]
```diff
@@ -140,6 +142,52 @@ class TimedataFormat:
...
+            stratigraphic_alias=[]
+            if no_start_or_missing_name
+            else strat[name].get("stratigraphic_alias"),
+            offset=None if no_start_or_missing_name else strat[name].get("offset"),
+            top=None if no_start_or_missing_name else strat[name].get("top"),
+            base=None if no_start_or_missing_name else strat[name].get("base"),
+        )
+
+        if not no_start_or_missing_name and rv.name != "name":
+            rv.alias.append(name)
```
Comment (janbjorge) [inline src/fmu/dataio/_objectdata_provider.py:186]: Can be moved inside the above function block by usage of walrus op. But gets unreadable.

File: src/fmu/dataio/_objectdata_provider.py [hunk 3]
```diff
@@ -155,39 +170,33 @@ class _ObjectDataProvider:
...
+    bbox: dict = field(default_factory=dict)
+    classname: str = field(default="")
+    efolder: str = field(default="")
+    extension: str = field(default="")
+    fmt: str = field(default="")
+    layout: str = field(default="")
+    metadata: dict = field(default_factory=dict)
+    name: str = field(default="")
+    specs: dict = field(default_factory=dict)
+    subtype: str = field(default="")
```
Comment (perolavsvendsen) [inline src/fmu/dataio/_objectdata_provider.py:187]: subtype...?

**Review by perolavsvendsen — APPROVED** [review]
As I read it, should be 1:1 with existing functionality for doing lookup in the stratigraphic column and replacing given name with derived actual name.

**Comment by janbjorge** [comment]
> As I read it, should be 1:1 with existing functionality for doing lookup in the stratigraphic column and replacing given name with derived actual name.

Correct, there should be no new logical changes here. Expect for that we now use a dataclass insted of a dict to communicate between functions/classes.

**Pull request merged**